cmd/runtimetest/main.go

```diff
@@ -409,10 +409,25 @@ func testReadAccess(path string) (readable bool, err error) {
 	if err != nil {
 		return false, err
 	}
-	if fi.Mode()&os.ModeType == 0 {
-		return testFileReadAccess(path)
+
+	// In case of a directory, we should check its readability in a special way.
+	// In other files, we should not check its Mode explicitly, because the runtime
+	// spec does not mandate the type of masked files. It could be a regular file
+	// or a character file (/dev/null), which is the case for runtimes like runc.
```

Comment:
I'd rather continue to test the mode.  For example, the runtime may use a permission change to make a symlink read-only (or not read/writable at all), and we don't want to use `testFileReadAccess` to test symlink readability.  I'd rather support `/dev/null` and other character devices with logic like:

```go
switch fi.Mode&os.ModeType {
case 0, os.ModeDevice|os.ModeCharDevice:
    return testFileReadAccess(path)
case os.ModeDir:
    return restDirectoryReadAccess(path)
}
return false, fmt.Errorf("cannot test read access for %q (mode %d)", path, fi.Mode())
```